### DISCUSSION TITLE
WSL2: git no xauth data

### DISCUSSION THREAD

**Discussion by divinebovine** [microsoft/WSL#6101] [General] [answered]
# Environment

```none
Windows build number: 19041.572
Your Distribution version: Ubuntu 20.04
Whether the issue is on WSL 2 and/or WSL 1: Linux version 4.19.104-microsoft-standard (oe-user@oe-host) (gcc version 8.2.0 (GCC)) #1 SMP Wed Feb 19 06:37:35 UTC 2020
```

# Steps to reproduce

1. git clone git@github.com:octocat/Hello-World.git

**WSL logs**: https://aka.ms/AA9z0a5

#  Expected behavior

```sh
Cloning into 'Hello-World'...
remote: Enumerating objects: 13, done.
remote: Total 13 (delta 0), reused 0 (delta 0), pack-reused 13
Receiving objects: 100% (13/13), done.
```

# Actual behavior

Results in:
```sh
Cloning into 'Hello-World'...
Warning: No xauth data; using fake authentication data for X11 forwarding.
X11 forwarding request failed on channel 0
remote: Enumerating objects: 13, done.
remote: Total 13 (delta 0), reused 0 (delta 0), pack-reused 13
Receiving objects: 100% (13/13), done.
```

**Comment by divinebovine** [comment]
I do have the following in my .bashrc
```bash
export LIBGL_ALWAYS_INDIRECT=1
export DISPLAY=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2}'):0.0
```
and it shows up in my env
```bash
env | grep -i 'display\|libgl'
LIBGL_ALWAYS_INDIRECT=1
DISPLAY=172.25.208.1:0.0
```

ssh config:
```bash
cat ~/.ssh/config
HOST github.com
  IdentityFile ~/.ssh/github_rsa
```

**Comment by therealkenc** [comment, marked as answer]
Generally will get better eyeballs on Linux config over at [serverfault](https://serverfault.com/questions/422908/how-can-i-prevent-the-warning-no-xauth-data-using-fake-authentication-data-for) or [stackoverflow](https://stackoverflow.com/questions/40056227/warning-no-xauth-data-using-fake-authentication-data-for-x11-forwarding/44100171) and the like.

**Reply by divinebovine** [reply]
Ok, I'll try there, I just did not expect WSL2 to attempt to use X11 forwarding with my given configuration, so it seemed like a bug. Thanks anyway.
